sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -15,59 +15,64 @@ export 'package:sky/src/painting/text_painter.dart';
 // differently. To work around this problem for now, we round fractional pixel
 // values up to the nearest whole pixel value. The right long-term fix is to do
 // layout using fixed precision arithmetic.
-double _applyFloatingPointHack(double layoutValue) {
+double applyFloatingPointHack(double layoutValue) {
   return layoutValue.ceilToDouble();
 }
 
 /// A render object that displays a paragraph of text
 class RenderParagraph extends RenderBox {
 
-  RenderParagraph(TextSpan text) : _textPainter = new TextPainter(text) {
+  RenderParagraph(TextSpan text) : textPainter = new TextPainter(text) {
     assert(text != null);
   }
 
-  TextPainter _textPainter;
+  // Exposed for use by subclasses.
+  TextPainter textPainter;
 
   BoxConstraints _constraintsForCurrentLayout; // when null, we don't have a current layout
 
   /// The text to display
-  TextSpan get text => _textPainter.text;
+  TextSpan get text => textPainter.text;
   void set text(TextSpan value) {
-    if (_textPainter.text == value)
+    if (textPainter.text == value)
       return;
-    _textPainter.text = value;
+    textPainter.text = value;
     _constraintsForCurrentLayout = null;
     markNeedsLayout();
   }
 
-  void _layout(BoxConstraints constraints) {
+  // Can be overridden by subclasses.
+  bool allowLineWrap() => true;
+
+  // Exposed for use by subclasses.
```

Comment:
I'd skip these comments.

Rather than exposing all these implementation details for `RenderParagraph`, maybe we should push this logic down into TextPainter.  It would be sensible for TextPainter's layout function to work based on a BoxConstraints rather than having separate values for each of the fields on BoxConstraints.  Depending on where BoxConstraints is implemented, we might need to push that whole object down into the painting layer, but that's fine.